test/make.sh

```diff
@@ -45,11 +45,20 @@ function docker() {
 # files ending in '.tf'
 function check_terraform() {
   echo "Running terraform validate"
-  #shellcheck disable=SC2156
-  find . -name "*.tf" -not -path "./autogen/*" -not -path "./test/fixtures/shared/*" -not -path "./test/fixtures/all_examples/*" -exec bash -c 'terraform validate --check-variables=false $(dirname "{}")' \;
+  #shellcheck disable=SC2038
+  find . -name "*.tf" \
+    -not -path "./autogen/*" \
+    -not -path "./test/fixtures/shared/*" \
+    -not -path "./test/fixtures/all_examples/*" \
+    | xargs dirname | sort | uniq | xargs -L 1 -i{} bash -c 'terraform init "{}" > /dev/null && terraform validate "{}"'
+
   echo "Running terraform fmt"
   #shellcheck disable=SC2156
-  find . -name "*.tf" -not -path "./autogen/*" -not -path "./test/fixtures/shared/*" -not -path "./test/fixtures/all_examples/*" -exec bash -c 'terraform fmt -check=true -write=false "{}"' \;
+  find . -name "*.tf" \
+    -not -path "./autogen/*" \
+    -not -path "./test/fixtures/shared/*" \
+    -not -path "./test/fixtures/all_examples/*" \
+    -exec bash -c 'terraform fmt -check=true -write=false "{}"' \;
```

Comment:
`terraform fmt` should only be run for each directory.

```suggestion
| xargs dirname | sort | uniq | xargs -L 1 -i{} bash -c 'terraform fmt "{}"'
```